tests/cephadm/test_package_validation.py

```diff
@@ -0,0 +1,162 @@
+import re
+
+from cli.utilities.containers import Container
+from cli.utilities.packages import Package, PackageError, ReposError, Rpm
+from cli.utilities.utils import get_custom_repo_url
+from utility.log import Log
+
+log = Log(__name__)
+
+
+YUM_REPO_DIR = "/etc/yum.repos.d"
+
+installed_packages = [
+    "chrony",
+    "lvm2",
+    "podman",
+]
+
+packages_to_install = [
+    "cephadm-ansible",
+]
+
+rpm_dependent_packages = [
+    "ansible-core",
+    "ansible-collection-ansible-posix",
+    "ansible-collection-community-general",
+]
+
+rpm_client_packages = [
+    "ceph-common",
+    "ceph-fuse",
+    "cephfs-top",
+]
+
+
+def validate_rpm_version(node, package_list, version_config):
+    """Validate installed rpm package version
+
+    Args:
+        node (ceph.ceph.CephNode): Node name
+        package_list (list): List of packages
+        version_config (dict): List of packages and package version
+    """
+    # Define regex to get version
+    rpm_regex = r"Version\s+:\s+(\d+\.\d+)"
+
+    # Check rpm package version using yum info
+    for pkg in package_list:
+        out = Package(node).info(pkg)
+        version = re.search(rpm_regex, out).group(1)
+        if version < version_config.get(pkg):
+            raise PackageError(
+                f"Expected version for {pkg} is {version_config.get(pkg)}; found {version}"
+            )
+        else:
+            log.info(f"{pkg} installed correctly with version {version}")
+
+
+def validate_image_version(node, image_dict, version_config):
+    """Validate image version
+
+    Args:
+        node (ceph.ceph.CephNode): Node name
+        image_dict (dict): List of images
+        version_config (dict): List of packages and package version
+    """
+    # Check container image version using podman inspect
+    for key, image in image_dict.items():
+        version = Container(node).inspect(image, format="'{{.Config.Labels.version}}'")[
+            0
+        ]
+        if version < version_config.get(key):
+            raise PackageError(
+                f"Expected version for {image} is {version_config.get(image)}; found {version}"
+            )
+        else:
+            log.info(f"{image} installed correctly with version {version}")
+
+
+def run(ceph_cluster, **kwargs):
+    """Verify package version validation
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    # Get installer and client node
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    client = ceph_cluster.get_nodes(role="client")[0]
+    log.info(f"Check type: {type(installer)}")
+
+    # Check if requied packages are installed
+    for pkg in installed_packages:
+        if not Rpm(installer).query(pkg):
+            raise PackageError(f"Required package {pkg} is not installed")
+
+    # Get config details
+    config = kwargs.get("config")
```

Comment:
Get these details at the start of the method

Reply:
I think this placement is correct at this step. 

As we are checking whether installed packages are present or not before getting the config information. If installed required packages are not present in the node these configs are useless. 

Please check again and let me know.

Reply:
All these are mandatory parameters, better to have validation at the start

Reply:
Done, updated.